libmachine/provision/custom_script.go

```diff
@@ -0,0 +1,35 @@
+package provision
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/rancher/machine/libmachine/provision/pkgaction"
+)
+
+func WithCustomScript(provisioner Provisioner, customScriptPath string) error {
+	if provisioner == nil {
+		return nil
+	}
+
+	if err := provisioner.SetHostname(provisioner.GetDriver().GetMachineName()); err != nil {
+		return err
+	}
+
+	for _, pkg := range provisioner.GetPackages() {
+		if err := provisioner.Package(pkg, pkgaction.Install); err != nil {
```

Comment:
Where do packages get set ?

Reply:
Depends on the provisioner, but most of them get set here: https://github.com/rancher/machine/blob/b5279793680d580e59f84ca4dcf6690fd1bcc930/libmachine/provision/systemd.go#L28-L30